deploy/operator/ztp/day2BaremetalHost.j2

```diff
@@ -0,0 +1,38 @@
+{% set ns = namespace(counter=1) %}
```

Comment:
This file is identical to the other `baremetalHost.j2` template except for the image section. Maybe we should just parametrized the other file?

Reply:
We can, although the other file has many conditions, and it will make it more complicated

Reply:
If we remove the image bit, then you won't need this template, right?

Reply:
Can change, but baremetalHost.j2 already has many conditions.  
- It iterates over masters only.
- It handles SNO differently

**_{% set ns = namespace(counter=1) %}
{% set bmhs = [] %}
{% if assisted_upgrade_operator|string == "true" %}
## Use the first baremetalhost to install SNO before upgrading the operator
      {{ bmhs.append( baremetalhosts[0] ) }}
{% elif ( assisted_stop_after_agent_discovery|string == "true" ) and ( baremetalhosts|length > 1) %}
## When operator has already been upgraded, use the other baremetalhost to install SNO after the operator upgrade
      {{ bmhs.append( baremetalhosts[1] ) }}
{% else %}
      {% set bmhs = baremetalhosts %}
{% endif %}
{% for host in bmhs %}
{% if ns.counter <= spoke_controlplane_agents|int %}_**